jax/abstract_arrays.py

```diff
@@ -101,9 +101,30 @@ def _canonicalize_dimension(dim):
   else:
     return operator.index(dim)
 
-def _canonicalize_shape(shape):
-  """Ensure shape is a tuple of int or registered _DIMENSION_TYPES."""
-  return tuple(map(_canonicalize_dimension, shape))
+def canonicalize_shape(shape):
+  """Canonicalizes and checks for errors in a user-provided shape value.
+
+  `None` shape is passed through as is.
+
+  Args:
+    shape: a Python value that represents a shape, or None.
+
+  Returns:
+    A tuple of integers, or None.
+  """
+  if shape is None:
```

Comment:
I think this part should probably be left out here? We don’t want ShapedArray instances for be able to have shape=None.

Reply:
I added it in response to the failures here: https://travis-ci.org/google/jax/jobs/646213098?utm_medium=notification&utm_source=github_status

I don't know if those use cases are valid or not, please advise.

Reply:
That's because the RNG functions accept `shape=None` to indicate "copy shape from the other arguments."

Instead of putting the `shape is None` check inside `canonicalize_dimension`, I would suggest including that check inline in each RNG function, e.g.,
```python
def uniform(key, shape=(), dtype=onp.float64, minval=0., maxval=1.):
  dtype = dtypes.canonicalize_dtype(dtype)
  if shape is not None:
    shape = abstract_arrays.canonicalize_shape(shape)
  return _uniform(key, shape, dtype, minval, maxval)
```

Reply:
Ok, did it as you suggest.